instruments/cryomagnetics/cryomagnetics_lm510.py

```diff
@@ -0,0 +1,193 @@
+        command = "CHAN %d" % channel
+        self.query(command)
+
+    @property
+    def status_byte(self):
+        """
+        Get a byte array representing the instrument's status byte. The
+        status byte describes whether channel 1 or 2 are ready to make a
+        measurement. The structure of the status byte is described in
+        greater detail in the level meter's operating manual.
+
+        :return: The current value of the status byte
+        :rtype: bytes
+        """
+        byte_as_string = self.query("*STB?")
+        return byte_as_string.encode('ascii')
+
+    @property
+    def channel_1_data_ready(self):
+        """
+
+        :return: ``True`` if channel 1 is ready to measure, otherwise ``False``
+        :rtype: bool
+        """
+        status_byte = self.status_byte
+        return (int(status_byte[0]) & 1) > 0
+
+    @property
+    def channel_2_data_ready(self):
+        """
+
+        :return: ``True`` if channel 2 is ready to measure, otherwise ``False``
+        :rtype: bool
+        """
+        status_byte = self.status_byte
+        return (int(status_byte[0]) & 4) > 0
+
+    @property
+    def channel_1_measurement(self):
+        """
+        Make a measurement on channel 1
+
+        :return: The level of the cryogen.
+        :rtype: Quantity
+        """
+        return self._measurement(1)
+
+    @property
+    def channel_2_measurement(self):
+        """
+        Make a measurement on channel 2
+
+        :return: The level of cryogen measured by the probe connected to
+            channel 2 in the meter
+        :rtype: Quantity
+        """
+        return self._measurement(2)
+
+    def reset(self):
+        """
+        Bring the instrument to a safe, known state
+        """
+        self.query("*RST")
+
+    def _measurement(self, channel_number):
+        """
+        Make a measurement on a particular channel
+
+        :param int channel_number: The channel on which the measurement is
+            to be made
+        :return: The measured value on that channel
+        :rtype: Quantity
+        """
+        measurer = self._ChannelMeasurement(channel_number, self)
+        response = measurer.measurement
+
+        return self.parse_response(response)
+
+    @staticmethod
+    def parse_response(response):
+        """
+        Extract the value and unit in which the response was returned,
+        and match these values to a physical quantity
+
+        :param str response: The response returned by making a measurement
+        :return: The measured quantity
+        :rtype: Quantity
+        """
+        value_match = re.search(r"^(\d|\.)*(?=\s)", response)
+        unit_match = re.search(r"(?<=\s).*(?=$)", response)
+
+        value = float(value_match.group(0))
+        unit = CryomagneticsLM510.UNITS[unit_match.group(0)]
+
+        return_value = value * unit
+
+        return return_value
+
+    class _ChannelMeasurement(object):
```

Comment:
As mentioned above, using a ``_Channel`` inner class might allow bringing the measurement into the same inner class as ``data_ready``, such that users could run ``inst.channel[0].measurement``. It may also be good to represent this as a method (``channel[0].measure()``) instead of a property to reflect that an action is being taken.